src/pykeen/pipeline/api.py

```diff
@@ -292,6 +280,18 @@ class PipelineResult(Result):
     #: How long in seconds did evaluation take?
     evaluate_seconds: float
 
+    #: The model trained by the pipeline
```

Comment:
Sorry I haven't been following discussion yet, but I'm pretty opposed to weakening the typing in this class. When people get results from the pipeline, they shouldn't have to check for if there are results there - it should be guaranteed.

This is definitely true for the model and seed. If the training triples don't get loaded properly, people get confused since they need to reference the training triples to do anything valuable. I don't want to make changes that will result in getting more github issues from confused users. Wrt the training loop, this could be a reasonable thing to make optional.

Reply:
Hi @cthoyt ! Yes, it's perfectly understandable. So, maybe we should give up the implmentation of this feature or rethink it completely if users need to reload the `PipelineResult`, since the current implementation of `save_to_directory` saves only partial information of the object.